sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1764,4 +1769,123 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     val df2 = spark.read.option("PREfersdecimaL", "true").json(records)
     assert(df2.schema == schema)
   }
+
+  test("SPARK-18352: Parse normal multi-line JSON files (compressed)") {
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+      primitiveFieldAndType
+        .toDF("value")
+        .write
+        .option("compression", "GzIp")
+        .text(path)
+
+      new File(path).listFiles() match {
+        case compressedFiles =>
+          assert(compressedFiles.exists(_.getName.endsWith(".gz")))
+      }
+
+      val jsonDF = spark.read.option("wholeFile", true).json(path)
+      val jsonDir = new File(dir, "json").getCanonicalPath
+      jsonDF.coalesce(1).write
+        .format("json")
+        .option("compression", "gZiP")
+        .save(jsonDir)
+
+      new File(jsonDir).listFiles() match {
+        case compressedFiles =>
+          assert(compressedFiles.exists(_.getName.endsWith(".json.gz")))
+      }
+
+      val jsonCopy = spark.read
```

Comment:
I'd like to write the whole-file json, then read it and check the answer, instead of writing and reading it again, which is too complicated.